tools/cache-zulip-git-version

```diff
@@ -2,4 +2,19 @@
 set -e
 
 cd "$(dirname "$0")/.."
-git describe --tags --match='[0-9]*' >zulip-git-version || true
+version=$(git describe --tags --match='[0-9]*') || "unknown"
+commit=$(git rev-parse HEAD) || "unknown"
+if git remote | grep -q "^upstream$"; then
+    merge_base_commit=$(git merge-base upstream/master HEAD) || "unknown"
+    fork_commits=$(git log --pretty=oneline "$merge_base_commit..HEAD" | wc -l) || "unknown"
```

Comment:
The main problem with this logic is that it does not consider the `upstream/3.x` type branches -- only upstream/master -- for finding the most appropriate merge base.